service/controller/azure/v5/resource/configmap/desired.go

```diff
@@ -0,0 +1,29 @@
+package configmap
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/cluster-operator/pkg/v5/configmap"
+	"github.com/giantswarm/cluster-operator/service/controller/azure/v5/key"
+)
+
+func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	configMapValues := configmap.ConfigMapValues{
+		ClusterID:    key.ClusterID(customObject),
+		Organization: key.ClusterOrganization(customObject),
+		WorkerCount:  key.WorkerCount(customObject),
+	}
+	desiredConfigMaps, err := r.configMap.GetDesiredState(ctx, configMapValues)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
```

Comment:
In the resource we get the desired state. Later we can add provider specific config maps here.